packages/sdk/src/types/runtime/appearance.ts

```diff
@@ -3,34 +3,99 @@
  * Licensed under the MIT License.
  */
 
-import { Actor, Material } from '.';
+import { Actor, GroupMask, Material } from '.';
 import { ZeroGuid } from '../../constants';
 
 export interface AppearanceLike {
-    enabled: boolean;
+    /**
+     * This actor's visibility preference, independent of its parent. If this property is a
+     * GroupMask object, this property will effectively be `true` for users in at least one
+     * of the groups, and `false` for everyone else. See [[GroupMask]].
+     */
+    enabled: boolean | GroupMask | number;
+
+    /**
+     * The ID of a previously-created [[Material]] asset.
+     */
     materialId: string;
 }
 
 export class Appearance implements AppearanceLike {
     /** @hidden */
-    public $DoNotObserve = ['activeAndEnabled', 'actor', 'material'];
+    public $DoNotObserve = ['actor', '_enabledFor'];
 
     // tslint:disable:variable-name
+    private _enabled = GroupMask.ALL_PACKED; // authoritative
+    private _enabledFor: GroupMask; // cached object, synced with _enabledPacked
     private _materialId = ZeroGuid;
     // tslint:enable:variable-name
 
-    /** Whether the actor with this appearance should be visible */
-    public enabled = true;
+    /**
+     * This actor's visibility preference, independent of its parent. See [[Appearance.activeAndEnabled]] for
+     * the computed visibility state. If this property is a GroupMask object, this property will
+     * effectively be `true` for users in at least one of the groups, and `false` for everyone else.
+     * See [[GroupMask]].
+     */
+    public get enabled(): boolean | GroupMask {
+        if (this.enabledPacked === GroupMask.ALL_PACKED) {
+            return true;
+        } else if (this.enabledPacked === GroupMask.NONE_PACKED) {
+            return false;
+        } else {
+            return this.enabledFor;
+        }
+    }
+    public set enabled(value: boolean | GroupMask) {
```

Comment:
I might be mistaken, but I think getters and setters each need their own doc headers. Grouping them won't make the documentation work for both.

Reply:
The header is for the property, and they're the same property. I've checked it, this works.

---
In reply to: [268862361](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/274#discussion_r268862361) [](ancestors = 268862361)